drivers/bluetooth/hci/slz_hci.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+
+#include <sl_btctrl_linklayer.h>
+#include <sl_hci_common_transport.h>
+#include <pa_conversions_efr32.h>
+#include <sl_bt_ll_zephyr.h>
+
+#define SL_BT_CONFIG_ACCEPT_LIST_SIZE	1
+#define SL_BT_CONFIG_MAX_CONNECTIONS	1
+#define SL_BT_CONFIG_USER_ADVERTISERS	1
+#define SL_BT_CONTROLLER_BUFFER_MEMORY  CONFIG_BT_SILABS_HCI_BUFFER_MEMORY
+#define SL_BT_SILABS_LL_STACK_SIZE	1024
+
+static K_KERNEL_STACK_DEFINE(slz_ll_stack, SL_BT_SILABS_LL_STACK_SIZE);
+static struct k_thread slz_ll_thread;
+
+void rail_isr_installer(void)
+{
+	IRQ_CONNECT(RDMAILBOX_IRQn, 0, RDMAILBOX_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_SEQ_IRQn, 0, RAC_SEQ_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_RSM_IRQn, 0, RAC_RSM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(PROTIMER_IRQn, 0, PROTIMER_IRQHandler, NULL, 0);
+	IRQ_CONNECT(MODEM_IRQn, 0, MODEM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(FRC_IRQn, 0, FRC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(BUFC_IRQn, 0, BUFC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(AGC_IRQn, 0, AGC_IRQHandler, NULL, 0);
+}
+
+uint32_t hci_common_transport_transmit(uint8_t *data, int16_t len)
```

Comment:
This function belongs to the OS abstraction layer? If so, please document the input and output ([using Doxygen comments](https://docs.zephyrproject.org/latest/contribute/bin_blobs.html#allowed-dependencies)).

Reply:
Thanks for spotting this. I've added the documentation.

Reply:
It looks to me like data[1] has a special meaning (HCI event code? Is there some authoritative documentation which could be linked to?), would mention this specifically.

(I mean, nowadays this is easy to figure out by reading the code, but what when bt_buf_get_evt() gets changed down the road. How can, whoever is tasked to maintaining this code and potentially not part of SiLabs, be sure in many years from now?)

Reply:
> It looks to me like data[0] and data[1] have special meanings, would mention those with some extra information (maybe refer to existing documentation?).

I don't think any additional documentation is needed here, as these are covered by the standard. The first byte is the packet type, and in event frames the second byte is for the event code (for LE events). I've changed the variable names to reflect that. Let me know if that's okay.

> (Sorry for being pedantic. Nowadays this is easy to figure out by reading the code, but what when the Zephyr functions get changed down the road. How can, whoever is tasked to maintaining this code and potentially not part of SiLabs, be sure in many years from now how exactly the blobs call this function with?)

No problem :) Thanks for looking at the code and reviewing it.

Reply:
> I don't think any additional documentation is needed here, as these are covered by the standard. The first byte is the packet type, and in event frames the second byte is for the event code (for LE events). I've changed the variable names to reflect that. Let me know if that's okay.

Reasoning seems fine to me. However, I'd describe param data as e.g. "Packet type followed by HCI packet data", as the first byte not actually gets transmitted as the current comment suggests.

Reply:
Thanks for your suggestion. I've updated the `param data` description.